ansible/roles/controller/tasks/deploy.yml

```diff
@@ -13,10 +13,10 @@
       "{{ (controller_index_base|int) + host_group.index(inventory_hostname) }}"
 
 - name: "pull the {{ docker.image.tag }} image of controller"
-  shell:
-    docker pull
-    {{ docker_registry~docker.image.prefix }}/controller:{{docker.image.tag}}
+  shell: "docker pull {{docker_registry}}{{ docker.image.prefix }}/controller:{{docker.image.tag}}"
   when: docker_registry != ""
+  register: result
+  until: (result.rc == 0)
```

Comment:
How do these retries interact with existing retries?

Reply:
I'm confuse, the task only has one entry for retires `retries: "{{ docker.pull.retries }}"`

Which other retries you are referring?